### PR TITLE
Add name option to makeStyles

### PR DESCRIPTION
**Pull request by DanudeSandstorm**
I've begun adding the name of the component in the ra-ui-materialui components that use ``makeStyles`` as mentioned in #3701 .
Please let me know whether this is what is expected and I can perform this addition on the remaining components. 

Note: There's a difference in prettier/linting specification between ``.tsx`` and ``.js`` files, which results in differences in the makeStyles formatting.

### CODE REVIEW

File: packages/ra-ui-materialui/src/auth/Logout.tsx [hunk 1]
```diff
@@ -13,12 +13,15 @@ interface Props {
...
-    icon: { minWidth: theme.spacing(5) },
-}));
+const useStyles = makeStyles(
+    (theme: Theme) => ({
+        menuItem: {
+            color: theme.palette.text.secondary,
+        },
+        icon: { minWidth: theme.spacing(5) },
+    }),
+    { name: 'LogoutWithRef' }
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/auth/Logout.tsx]: Why not Logout?

Comment (DanudeSandstorm) [reply packages/ra-ui-materialui/src/auth/Logout.tsx]: I was going by the export default. But I'll change it to the filename instead.

File: packages/ra-ui-materialui/src/detail/TabbedShowLayout.js [hunk 2]
```diff
@@ -21,13 +21,16 @@ const sanitizeRestProps = ({
...
-}));
+const useStyles = makeStyles(
+    theme => ({
+        content: {
+            paddingTop: theme.spacing(1),
+            paddingLeft: theme.spacing(2),
+            paddingRight: theme.spacing(2),
+        },
+    }),
+    { name: 'TabbedShow' }
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/detail/TabbedShowLayout.js]:
```suggestion
    { name: 'TabbedShowLayout' }
```

**Comment by WiXSL** [comment]
@DanudeSandstorm can you rebase?

**Comment by DanudeSandstorm** [comment]
I rebased onto next

**Comment by WiXSL** [comment]
Thank you.
Cause I need those changes for styling as well.

**Comment by WiXSL** [comment]
@fzaninotto , is this PR ready to be merged?

**Comment by fzaninotto** [comment]
Excellent, thanks!

**Pull request merged**